### ISSUE TITLE
unpredictable if component.attrValue or component.data is set

### ISSUE THREAD

**Issue by asandersSAP** [aframevr/aframe#4078]
**Description:**
It's unpredictable to guess whether the component.attrValue or the component.data contains the updated value. On components.material.data for example I get "undefined" where material.data works. On any custom component for example I have to use component.data. On other randomly chosen components sometimes its .data and sometimes .attrValue

Since components.attrValue works always with sometimes a additional .data field we could say to use attrValue always. The thing is that it contains strings always. 
So if I have clauses like if(el.components.material.attrValue) it's always true because the string only needs to exist. 

Long story short - in 0.5.0 you listed this as a FIX: 
Fixed component updates with setAttribute using data instead of attrValue.

So I guess .data should always be the updated and existing version since attrValue is deprecated and this one is a bug? 

- A-Frame Version: 0.9.0
- Platform / Device: MacOS Mojave // all browsers

**Comment by ngokevin** [comment]
They are both internal properties. Use `.getAttribute` to read data.
